src/components/Tag/Tag.scss

```diff
@@ -76,6 +76,7 @@ $icon-size: rem(16px);
   overflow: hidden;
   text-overflow: ellipsis;
   white-space: nowrap;
+  vertical-align: middle;
```

Comment:
I'm not sure why align-items wouldn't be working for some use cases. There doesn't seem to be any difference when comparing your playground code in this branch against master. This seems like a simple and safe enough change but I'm curious what's causing align-items not to work in your image

Reply:
hey @kyledurand! thanks for taking a look! 

so, in our consuming project, our tag-related modals were breaking when there was a super long tag. to fix this issue, i wrapped the `<Tag/>` in a div to apply styling to it, with a `max-width`, `display: inline-block`, `overflow: hidden`, `text-overflow: ellipsis`, etc. but after wrapping the `<Tag/>` in the div, i saw that the vertical centering of the tag text broke. 

i connected this branch with the consuming project, and it fixed the vertical alignment. :)